### PR TITLE
WB-1603: Implement PhosphorIcon component

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:

This PR adds support for the `@phosphor-icons/core` package to `wonder-blocks-icon`.

Adds a new `PhosphorIcon` component that wraps a `span` element that
loads the SVG file imported from `@phosphor-icons/core`.

Issue: WB-1590, WB-1603

## Test plan:

Verify that the `PhosphorIcon` stories render correctly.

/?path=/docs/icon-phosphoricon--docs

Screenshot 2023-09-25 at 4 49 14 PM

### CODE REVIEW

File: __docs__/wonder-blocks-icon/phosphor-icon.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+ * A `PhosphorIcon` displays a small informational or decorative image as an
+ * HTML element that renders a Phosphor Icon SVG available from the
+ * `@phosphor-icons/core` package.
+ *
+ * For more information about the icons catalog, check the [Phosphor Icons
+ * website](https://phosphoricons.com/).
+ *
+ * ## Usage
+ *
+ * ```js
```
Comment (somewhatabstract) [inline __docs__/wonder-blocks-icon/phosphor-icon.stories.tsx:25]: **nit:**
```suggestion
 * ```tsx
```

File: packages/wonder-blocks-icon/src/components/__tests__/phosphor-icon.test.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            () => expectedRenderSize,
+        );
+
+        // Act
+        render(<PhosphorIcon icon={Plus} testId="phosphor-icon" />);
+
+        // Assert
+        const icon = screen.getByTestId("phosphor-icon");
+        expect(icon).toHaveStyle(`width: ${expectedRenderSize}px`);
+        expect(icon).toHaveStyle(`height: ${expectedRenderSize}px`);
```
Comment (somewhatabstract) [inline packages/wonder-blocks-icon/src/components/__tests__/phosphor-icon.test.tsx:73]: **nit:** Two expectations implies the need for two separate test cases to address two separate requirements. For example, "sets image height" and "sets image width".

File: packages/wonder-blocks-icon/src/components/phosphor-icon.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+ * A `PhosphorIcon` displays a small informational or decorative image as an
+ * HTML element that renders a Phosphor Icon SVG available from the
+ * `@phosphor-icons/core` package.
+ *
+ * For more information about the icons catalog, check the [Phosphor Icons
+ * website](https://phosphoricons.com/).
+ *
+ * ## Usage
+ *
+ * ```js
```
Comment (somewhatabstract) [inline packages/wonder-blocks-icon/src/components/phosphor-icon.tsx:96]: **nit:**
```suggestion
 * ```tsx
```

File: packages/wonder-blocks-icon/src/components/phosphor-icon.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+ *     color={Color.blue}
+ *     size="medium"
+ *     style={{margin: Spacing.xxxxSmall_2}}
+ * />
+ * ```
+ *
+ * These icons should fit into a viewport of `16`, `24`, `48`, and `96` pixels,
+ * respectively.
+ */
+const PhosphorIcon = React.forwardRef(function PhosphorIcon(
```
Comment (somewhatabstract) [inline packages/wonder-blocks-icon/src/components/phosphor-icon.tsx:111]: **nit:** Prefer named exports?
```suggestion
export const PhosphorIcon = React.forwardRef(function PhosphorIcon(
```

File: packages/wonder-blocks-icon/src/components/phosphor-icon.tsx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        display: "inline-block",
+        verticalAlign: "text-bottom",
+        flexShrink: 0,
+        flexGrow: 0,
+    },
+});
+
+PhosphorIcon.displayName = "PhosphorIcon";
+
+export default PhosphorIcon;
```
Comment (somewhatabstract) [inline packages/wonder-blocks-icon/src/components/phosphor-icon.tsx:162]: **nit:** Prefer named exports and make the earlier change, removing this line?

File: packages/wonder-blocks-icon/src/index.ts [hunk 6]
```diff
@@ -2,6 +2,7 @@ import Icon from "./components/icon";
 import type {IconAsset, IconSize} from "./util/icon-assets";
 
 export * as icons from "./util/icon-assets";
-
+export {default as PhosphorIcon} from "./components/phosphor-icon";
```
Comment (somewhatabstract) [inline packages/wonder-blocks-icon/src/index.ts:5]: **nit:** If this was a named export, you can just do:
```suggestion
export {PhosphorIcon} from "./components/phosphor-icon";
```

File: packages/wonder-blocks-icon/src/types.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+/**
+ * All the possible icon weights.
+ */
+export type PhosphorIconAsset = PhosphorRegular | PhosphorBold | PhosphorFill;
+/**
+ * The different icon weights per size.
+ */
+export type PhosphorIconSmall = PhosphorBold | PhosphorFill;
+export type PhosphorIconMedium = PhosphorRegular | PhosphorFill;
```
Comment (somewhatabstract) [inline packages/wonder-blocks-icon/src/types.ts:9]: **nit:** IDE's will label `PhosporIconSmall` with this comment, and then have no comment for `PhosphorIconMedium`. Give each its own comment.
```suggestion
/**
 * The different icon weights for small icons.
 */
export type PhosphorIconSmall = PhosphorBold | PhosphorFill;
/**
 * The different icon weights for medium icons.
 */
export type PhosphorIconMedium = PhosphorRegular | PhosphorFill;
```

**Review by somewhatabstract — APPROVED** [review]
Generally looks good to me. A few nits inline to consider.

---
My review uses [Conventional Comments][1] to add context and set expectations for the comments I am leaving.

[1]:https://conventionalcomments.org/

File: packages/wonder-blocks-icon/src/components/phosphor-icon.tsx (hunk 4 above)
Comment (jandrade) [reply packages/wonder-blocks-icon/src/components/phosphor-icon.tsx:111]: Fair enough :)

File: packages/wonder-blocks-icon/src/index.ts (hunk 6 above)
Comment (jandrade) [reply packages/wonder-blocks-icon/src/index.ts:5]: you already sold me in the previous comment! thanks for the suggestion.

File: packages/wonder-blocks-icon/src/types.ts (hunk 7 above)
Comment (jandrade) [reply packages/wonder-blocks-icon/src/types.ts:9]: ah great point! will make the change.

**Pull request merged**